### PR TITLE
feat: new /schedule API endpoint

### PR DESCRIPTION
**Pull request by lemald**
Asana ticket: [⚙️ Minischedules showing incorrect run with mid-route swing](https://app.asana.com/0/1200180014510248/1200317928103334/f)

Splitting just the back-end change out so that we can deploy it before deploying the front-end code that depends on it. See full commit message for details on the approach.

### CODE REVIEW

File: lib/skate_web/controllers/schedule_conroller.ex [hunk 1]
```diff
@@ -0,0 +1,29 @@
+defmodule SkateWeb.ScheduleController do
```
Comment (phildarnowsky) [inline lib/skate_web/controllers/schedule_conroller.ex:1]: **nitpick:** typo in filename ("conroller")

File: test/schedule/data_test.exs [hunk 2]
```diff
@@ -645,7 +645,74 @@ defmodule Schedule.DataTest do
     end
   end
 
-  describe "block_for_trip/1" do
+  describe "run_for_trip/3" do
+    test "returns run with ID for trip" do
+      trip = %Schedule.Trip{
```
Comment (phildarnowsky) [inline test/schedule/data_test.exs:650]: **suggestion:** let's use factories here. and throughout

**Review by phildarnowsky — COMMENTED** [review]
**praise:** overall solid! just a few minor changes requested.

**Comment by lemald** [comment]
@phildarnowsky updated based on your feedback. I'll get this merged once you approve and then rebase the other PR.

**Review by phildarnowsky — APPROVED** [review]

**Pull request merged**